EShop.Web/Controllers/ProductsController.cs

```diff
@@ -36,5 +36,30 @@ public IActionResult PostProduct()
         {
            return this.View();
         }
+
+        public IActionResult DeleteProduct(int id) 
+        {
+            this.productRepository.Remove(id);
+
+            return this.RedirectToAction("Index");
+        }
+        
+        [HttpPost]
+        [ValidateAntiForgeryToken]
+        public IActionResult EditProduct(int id ,ProductModel product)
+        {
+            if(this.ModelState.IsValid && product.Id == id) 
```

Comment:
Cand scri functii, si ai if-uri, e mai bine sa faci urmatoarea schema:

```
     if (ConditiaA && ConditiaB)
     {
         return Ceva;
     } else {
         return Altceva;
     }
```

In cazul de sus, e ideal sa separi conditiile in puncte de iesire:

```
     if (ConditieA == false) {
         return Altceva; // lipseste ConditiaA 
     }
     
     if (ConditieB == false) {
         return Altceva; // lipseste ConditiaB
     }
     
     return Ceva; // toate celelalte cazuri sunt deja tratate, ai tot ce iti trebuie aici
     
     ```

Reply:
Drept urmare

```if (this.ModelState.IsInvalid) { return this.View() } // neaga conditia #1```
```if (product.Id != id) { return this.View() } // neaga conditia #2```

La finalul functiei ai fara else-uri cazul fericit 😄 
```
      this.productRepository.Update(product, id);
      return this.RedirectToAction("Index");
```

Reply:
Asa, cand citesti codul, sti care cazuri sunt 'invalide' si care e intentia functiei la finalu zilei (ultima clauza)